### DISCUSSION TITLE
Best practice to use/deploy configuration for static SPA

### DISCUSSION THREAD

**Discussion by hlassiege** [nuxt/nuxt#16015] [Questions]
Hi,

I'd like to know what would be the best practice, if any, when using configuration for static SPA.

For example, let's say I want to configure an API with an API key that could vary with the environment (integration, production).

I know I can use runtimeConfig :

``` 
    runtimeConfig: {
        public: {
            myapi: {
                public_key: process.env.API_KEY || '',
            },
        },
    },
```

However, it means I have to build my app several times, one for each environment. 

Example : 
```
   export API_KEY=toto
   pnpm run generate
   export API_KEY=titi
   pnpm run generate
```

Another option would be to load the configuration at runtime, as discussed here : 
https://github.com/nuxt/framework/discussions/5061

Another option would be, at deploy time, to search and replace some strings in each html file 

In this case, I'd configure the file like this : 

``` 
    runtimeConfig: {
        public: {
            myapi: {
                public_key: process.env.API_KEY ||  'API_KEY_TO_REPLACE_AT_DEPLOY_TIME',
            },
        },
    },
```

By default, in the CI, it will generate static HTML files with API_KEY_TO_REPLACE_AT_DEPLOY_TIME
At deploy time, I can search and replace in every html file. Ugly ? Maybe. 

And you, what are your practices ?

**Comment by EinBexiii** [comment]
Have you maybe found out a better solution instead of replacing each time if it's generated? I am currently wondering how best to implement it. Since I am currently programming a website for a client, and want to have a config where they can quickly change the title, for example.
